sklearn/utils/testing.py

```diff
@@ -398,6 +405,72 @@ def assert_raise_message(exception, message, function, *args, **kwargs):
         assert_in(message, error_message)
 
 
+def assert_not_raises(given_exception, function, *args, **kwargs):
+    """Helper function to make sure that the given error was not raised"""
+
+    try:
+        function(*args, **kwargs)
+    except Exception as e:
+        if isinstance(e, given_exception):
+            raise AssertionError("%r was raised" % e)
+
+
+def _assert_same_model_method(method, X, estimator1, estimator2, msg=None):
+    if hasattr(estimator1, method):
+        m = '%r\n\nhas %s, but\n\n%r\n\ndoes not' % (estimator1, 
+                                                     method,
+                                                     estimator2)
+        if not hasattr(estimator2, method):
+            raise AttributeError(m)
+ 
+        # Check if the method(X) returns the same for both models.
+        same_model = np.allclose(getattr(estimator1, method)(X), 
+                                 getattr(estimator2, method)(X))
+
+            
+        if not same_model:
+            if msg is None:
+                msg = ("Models are not equal. \n\n%s method returned"
+                       " different results for :\n\n%r \n\nand :\n\n%r" % 
+                       (method, estimator1, estimator2))
+
+            raise AssertionError(msg)
+
+
+def _assert_not_same_model_method(method, X, estimator1, estimator2, msg=None):
+    if hasattr(estimator1, method):
+        m = '{!r}\n\nhas {!s}, but\n\n{!r}\n\ndoes not'.format(estimator1, 
+                                                               method,
+                                                               estimator2)
+        # Even though we check for different models,
+        # AssertionError is raised if method present only in estimator1
+        if not hasattr(estimator2, method):
+            raise AttributeError(m)
+       
+        # Check if the method(X) returns the same for both models.
+        same_model = np.allclose(getattr(estimator1, method)(X), 
+                                 getattr(estimator2, method)(X))
+
+        if same_model:
+            if msg is None:
+                msg = ("Models are equal. \n\n%s method returned"
+                       " same results for :\n\n%r \n\nand :\n\n%r" % 
+                       (method, estimator1, estimator2))
+            raise AssertionError(msg)
+
+
+def assert_same_model(X, estimator1, estimator2, msg=None):
```

Comment:
I'm not sure if this is a good way to check whether two models are the same. ``predict`` might give the same result for very different models.
Why not check if the estimated attributes are close?
Or at least check the decision functions, too?